src/components/scenes/existingAccout/ChangeRecoveryScene.tsx

```diff
@@ -1,523 +1,459 @@
+    setQuestion1(questionPrompt)
+    setQuestion2(questionPrompt)
+  }
+  const handleNext = () => {
+    const errors = [errorAnswer1, errorAnswer2]
+    if (
+      question1 === questionPrompt ||
+      question2 === questionPrompt ||
+      errors.some(error => error != null)
+    ) {
       return
     }
     Keyboard.dismiss()
-    this.enableRecovery().catch(showError)
-  }
-
-  handleQuestion1 = () => {
-    this.setState({
-      showQuestionPicker: true,
-      focusFirst: true,
-      focusSecond: false
-    })
+    setAwaitingConfirmation(true)
   }
 
-  handleQuestion2 = () => {
-    this.setState({
-      showQuestionPicker: true,
-      focusFirst: false,
-      focusSecond: true
-    })
-  }
-
-  handleAnswer1 = (arg: string) => {
-    this.setState({
-      answer1: arg
+  const handleQuestion1 = () => {
+    Airship.show<string | undefined>(bridge => (
+      <RadioListModal
+        bridge={bridge}
+        title={s.strings.recovery_question_one}
+        items={question1pool}
+        selected={question1}
+      />
+    )).then(question => {
+      if (question == null) return
+      setQuestion1(question)
+      setAnswer1(answerPrompt)
     })
   }
 
-  handleAnswer2 = (arg: string) => {
-    this.setState({
-      answer2: arg
+  const handleQuestion2 = () => {
+    Airship.show<string | undefined>(bridge => (
+      <RadioListModal
+        bridge={bridge}
+        title={s.strings.recovery_question_one}
+        items={question2pool}
+        selected={question2}
+      />
+    )).then(question2 => {
+      if (question2 == null) return
+      setQuestion2(question2)
+      setAnswer2(answerPrompt)
     })
   }
 
-  handleQuestionSelected = (data: EdgeRecoveryQuestionChoice) => {
-    const question = data.question
-    if (this.state.focusFirst) {
-      this.setState({
-        question1: question,
-        showQuestionPicker: false
-      })
-      return
-    }
-    this.setState({
-      question2: question,
-      showQuestionPicker: false
+  const showDatePickerModal = (
+    setDate: (date: string) => void,
+    setError: (error: string | null) => void
+  ) => {
+    Airship.show<Date | undefined>(bridge => (
+      <DateModal bridge={bridge} />
+    )).then(answer => {
+      if (answer == null) return
+      setDate(toLocalDate(answer))
+      setError(null)
     })
   }
 
-  async enableRecovery(): Promise<void> {
-    const buttons: { [key: string]: ButtonInfo } = {
-      email: {
-        label: s.strings.confirm_email,
-        onPress: this.handleEnableEmail
-      },
-      share: {
-        label: s.strings.confirm_share,
-        onPress: this.handleEnableShare
-      },
-      cancel: { label: s.strings.cancel, type: 'secondary' }
+  const showTextInputModal = async (
+    title: string,
+    question: EdgeRecoveryQuestionChoice,
+    setText: (answer: string) => void,
+    setError: (error: string | null) => void
+  ) => {
+    const onSubmit = async (answer: string): Promise<true | string> => {
```

Comment:
Should be called `handleSubmit`. We always want `onX={handleY}` as a prop. There is an ESLint rule to enforce this, and although we haven't enabled that rule yet, we do want to follow it.

Reply:
Ok got it. Side note; is it ok that this is a handler within a function?